homeassistant/components/mobile_app/webhook.py

```diff
@@ -314,3 +315,8 @@
 
         return webhook_response(resp, registration=registration,
                                 headers=headers)
+
+    if webhook_type == WEBHOOK_TYPE_DELETE_REGISTRATION:
+        await delete_webhook(hass, config_entry)
```

Comment:
In this case you want to delete more than just the webhook, no? Take down all entities too for this device?